demo/core_14_$.utils.html.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Code City: HTML Utilities.
+ *
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview HTML utilities for Code City.
+ * @author fraser@google.com (Neil Fraser)
+ */
+
+$.utils.html = {}
+
+$.utils.html.escape = function(text) {
+  // Escape text so that it is safe to print as HTML.
+  return String(text).replace(/&/g, '&amp;').replace(/"/g, '&quot;')
+      .replace(/</g, '&lt;').replace(/>/g, '&gt;');
+};
+
+$.utils.html.preserveWhitespace = function(text) {
+  // Escape text so that it is safe and preserves whitespace formatting as HTML.
```

Comment:
I think it might be worth adding a bit more explanation here.  I was puzzling about the two `/  /g`s below, wondering if there were some characters in them which were not displaying (e.g., if they were different kinds of whitespace characters or something).  And worse: I had already looked at this code before, and figured it out, and *still* didn't immediately understand what was going on.

Reply:
Comment added:
Runs of three spaces ('   ') need to be escaped twice ('_  ', '__ ').